sycl/source/detail/device_image_impl.hpp

```diff
@@ -1053,7 +1053,8 @@ class device_image_impl {
       auto NewSupportingDevsEnd = std::remove_if(
           SupportingDevs.begin(), SupportingDevs.end(),
           [&NewImageRef](const sycl::device &SDev) {
-            return !doesDevSupportDeviceRequirements(SDev, NewImageRef);
+            auto &device_impl = *detail::getSyclObjImpl(SDev);
+            return !doesDevSupportDeviceRequirements(device_impl, NewImageRef);
```

Comment:
```suggestion
            return !doesDevSupportDeviceRequirements(*detail::getSyclObjImpl(SDev),
                                                     NewImageRef);
```

`auto` should be used when the type is either obvious (like `std::make_shared<...>`) or is too obfuscated and unimportant (like iterators). This isn't one of this scenarios, and to avoid the hardest problem in the programming (naming, as `device_impl` is both a variable and a type) it's just easier to inline this :)